crates/cairo-profiler/src/trace_reader/function_trace_builder.rs

```diff
@@ -220,3 +251,62 @@ fn build_current_call_stack(
         current_call_stack
     }
 }
+
+fn stack_trace_to_samples(
+    functions_stack_traces: HashMap<Vec<FunctionCall>, Steps>,
+    os_resources_map: &OsResources,
+) -> Vec<Sample> {
+    functions_stack_traces
+        .into_iter()
+        .map(|(call_stack, steps)| {
+            let mut measurements: HashMap<MeasurementUnit, MeasurementValue> = vec![(
+                MeasurementUnit::from("steps".to_string()),
+                MeasurementValue(i64::try_from(steps.0).unwrap()),
+            )]
+            .into_iter()
+            .collect();
+
+            if let Some(FunctionCall::InternalFunctionCall(InternalFunctionCall::Syscall(
+                function_name,
+            ))) = call_stack.last()
+            {
+                let Ok(syscall) = map_syscall_name_to_selector(function_name.0.as_str()) else {
+                    // todo: print the error in debug mode
+                    return Sample {
+                        call_stack,
+                        measurements,
+                    };
+                };
+                let resources = os_resources_map
+                    .execute_syscalls
+                    .get(&syscall)
+                    .unwrap_or_else(|| {
+                        panic!("Missing syscall {syscall:?} from versioned constants file")
```

Comment:
We should validate if all syscalls' costs are present in the versioned constants file when reading it so we can unwrap safely here

Reply:
What would be the upside of this approach? I get users might fiddle with the file so technically this might be viewed as user error, but generally speaking if it's missing (and they do not pass a custom map), it's our fault right?

Reply:
If they pass a custom map with missing syscalls: it is a user error and we should report it as early as possible (during reading of the file)

If it is missing at this point in the code it is our fault, that's why unwrapping here is "safe" (if it fails i means we don't handle a certain syscall)

Reply:
Not necessarily, check here: https://github.com/software-mansion/cairo-profiler/pull/112#discussion_r1819562774

Reply:
I have modified things a bit, now we always know given syscall is there because we validate it in versioned constants file reader. We also do not hold the list in several places, there is only one mapping function in stack_trace file that operated on types.